src/Corax/Utils/EntryIdEncodings.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+namespace Corax.Utils;
+
+public static class EntryIdEncodings
+{
+    //Schema:
+    //[CONTAINER_ID][FREQUENCY][CONTAINER_TYPE]
```

Comment:
`[id - 54 bits][freq - 8 bits][type - 2 bits]` - right?

Reply:
Yes. But I assume we can go further with optimizations. Since two bits of container id are empty (we used to encode container type there, I'm right?) we can push frequency into that. This would give us additional 2 bits of space for container id. Isn't?

Reply:
How ?
You need to store the container id in there anyway, how would you be able to ignore that?

Reply:
Before we didn't shift entry id to encode the container id, I assume that two bits are empty by default
https://github.com/ravendb/ravendb/blob/1948ee11fa76ee1d31c315eb0270e69bbdfe7167/src/Corax/IndexWriter.cs#L1449-L1452

So we can do it like this:
Encode:
`entryId << 8 | frequency << 2 | containerType`

Then on decode:
value (<- our encoded entry)
entryId:
`(value >> 8) & ~0b11`
container:
`value & 0b11`
frequency:
`(value >> 2 ) & 0xFF`

But I'm not 100% sure. If yes we can increase the number of entries up to `2^56` :)

Reply:
> So yes, you are correct that we can go to 2^56, however, that isn't the value that we are getting from the Container, so you'll need to shift it too. I would rather have a 2^54 limit (the number is big enough) and if needed, we can extend that later. Note that later is probably something our great-great-grandkids will need.

Call me old fashion here, but I would like to have a suitable variation of this comment baked in the code to explain the rationale and prevent our grandkids to try to fix it. :D